landingai/pipeline/frameset.py

```diff
@@ -79,6 +80,52 @@ class Frame(BaseModel):
     metadata: Dict[str, Any] = {}
     """An optional collection of metadata"""
 
+    @property
+    def frames(self) -> List["Frame"]:
+        """Returns a list with a single frame"""
+        warnings.warn(
+            "frame.frames[x].<method> is deprecated and will be removed in future versions. "
+            "Use frame.<method> instead"
+        )
+        return [self]
+
+    @classmethod
+    def from_image(cls, uri: str, metadata: Optional[Dict[str, Any]] = {}) -> "Frame":
+        """Creates a Frame from an image file
+
+        Parameters
+        ----------
+        uri : URI to file (local or remote)
+
+        Returns
+        -------
+        Frame : New Frame enclosing the image
+        """
+
+        image = Image.open(str(fetch_from_uri(uri)))
+        return cls(image=image, metadata=metadata)
+
+    @classmethod
+    def from_array(cls, array: np.ndarray, is_bgr: bool = True) -> "Frame":
```

Comment:
for image array could we enforce the correct type we expect, e.g.:
```
from numpy.typing import NDArray
@classmethod
def from_array(cls, array: NDArray[(Any, Any, 3), UInt8], is_bgr: bool = True) -> "Frame":
    ...
```
Also maybe an assert to check the array dimension, e.g.:
assert array.ndim == 3, "The image should have 3 dimensions (HWC)"

Reply:
Let me do that in a separated PR, just to limit a bit the amount of changes in this one...